lib/chef/knife/bootstrap/common_bootstrap_options.rb

```diff
@@ -50,6 +50,12 @@ def self.included(includer)
               :proc => lambda { |o| o.split(/[\s,]+/) },
               :default => []
 
+            option :environment,
+              :short        => "-E ENVIRONMENT",
+              :long         => "--environment ENVIRONMENT",
+              :description  => "Set the Chef environment (except for in searches, where this will be flagrantly ignored)",
+              :default      => "_default"
```

Comment:
do we have to set the default to `_default` here? We don't in `chef/lib/application/knife.rb`. I'm afraid this makes us dependent on this always being the default on the remote end as well.

Reply:
Yes, we need to set the `_default` here otherwise value of the `environment` sets as nil here https://github.com/chef-partners/azure-chef-extension/blob/master/lib/chef/azure/commands/enable.rb#L259 which causes `run-list` failure. 

Other way to fix this is in extension side as I did here https://github.com/MsysTechnologiesllc/azure-chef-extension/commit/eeb8b15c7ba32feb6fe01af936a6011b043a8d9d.

Reply:
@dheerajd-msys I have verified with adding recipe in my knife azurerm server create command and my recipe is executing properly without above code changes. I fill we need to debug the exact issue for the user.  There are two possibilities the user cookbook has some issues or cookbook is not uploaded on chef server.

Reply:
@Vasu1105 Yes, I have also verified by running `knife azurerm create server` on another machine that cookbook/recipe is executing properly without any code changes. So this PR is no longer required and can be closed.